pkg/nodeconfig/nodeconfig.go

```diff
@@ -26,7 +26,7 @@ const (
 func getNodeArgs() (string, error) {
 	nodeArgsList := []string{}
 	for _, arg := range os.Args[1:] {
-		if strings.Contains(arg, "=") {
+		if strings.HasPrefix(arg, "--") && strings.Contains(arg, "=") {
```

Comment:
note: at one point I was going to do the setproctitle here, using the existing secret redaction logic, but it did not work well. While fixing this I noted that the arg splitting was incorrectly splitting pass-through arguments such as `--kubelet-arg max-pods=100`. If you pass this in a single argv entry by using `--kubelet-arg=max-pods=100` it is split correctly, but we should not be splitting flag values that are passed as separate argv entries.